### PR TITLE
JAMES-3184 Limit concurrency while generating throttled entries

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
> I'm ok with the short term fix, not the long term 

Could you provide a viable alternative? (Or at least explain what's wrong with it?)

> but we don't care for now.

That depends if we reach the 1000 msg per second target, no?

File: mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReIndexerPerformer.java [hunk 1]
```diff
@@ -172,7 +174,7 @@ public ReIndexerPerformer(MailboxManager mailboxManager,
 
         try {
             Flux<Either<Failure, ReIndexingEntry>> entriesToIndex = mailboxMapper.findMailboxWithPathLike(mailboxQuery.asUserBound())
-                .flatMap(mailbox -> reIndexingEntriesForMailbox(mailbox, mailboxSession));
+                .flatMap(mailbox -> reIndexingEntriesForMailbox(mailbox, mailboxSession), 1);
```
Comment (rouazana) [inline mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReIndexerPerformer.java:177]: still a magic value?

**Comment by chibenwa** [comment]
test this please (build rescheduled)

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
```
[9b620da998a930295a712b043878957e177362e9] [ERROR] Errors: 
[9b620da998a930295a712b043878957e177362e9] [ERROR]   ConsistencyTasksIntegrationTest.shouldRecomputeMailboxCounters:268 » ConditionTimeout
```
test this please

**Comment by chibenwa** [comment]
ConsistencyTasksIntegrationTest.shouldRecomputeMailboxCounters:268 » ConditionTimeout

Unrelated

test this please

**Comment by chibenwa** [comment]
Merged

**Comment by mbaechler** [comment]
> > I'm ok with the short term fix, not the long term
> 
> Could you provide a viable alternative? (Or at least explain what's wrong with it?)

You wrote : 

> A long term solution would be to add a `concurrentMailboxes` parameter with the runningOptions.

We are back to giving advises to the server to prevent its collapse as far as I understand. I think a server should not collapse because of that.

The problem you are trying to solve, if I understand correctly is that you have "many mailbox queries" whose results are handled by a throttled processing.

This problem is supposed to be handled by back-pressure : your throttled process should ask for the right amount of mailboxes to have enough for its pipeline but prevent overflow.

Am I wrong?

**Comment by chibenwa** [comment]
```
This problem is supposed to be handled by back-pressure : your throttled process should ask for the right amount of mailboxes to have enough for its pipeline but prevent overflow.
```

This should do, yes.

**Pull request closed without merging**